core/src/main/java/org/apache/calcite/rel/RelNode.java

```diff
@@ -222,6 +223,22 @@
    */
   void explain(RelWriter pw);
 
+  /**
+   * Returns a relational expression string of this {@code RelNode}.
+   * The string returned is the same as
+   * {@link RelOptUtil#toString(org.apache.calcite.rel.RelNode)}.
+   *
+   * This method is intended mainly for use while debugging in an IDE,
+   * as a convenient short-hand for RelOptUtil.toString.
+   * We recommend that classes implementing this interface
+   * do not override this method.
+   *
+   * @return Relational expression string of this {@code RelNode}
+   */
+  default String explain() {
+    return RelOptUtil.toString(this);
```

Comment:
What is the difference between using `RelOptUtil.toString` and `RelOptUtil#dumpPlan` here?

Reply:
RelOptUtil#dumpPlan can dump other formats like XML/JSON, we don't need this here.